### PR TITLE
feat: Add tag batch-processing to XPath parser

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR allows to process tags contained in data in a batch fashion by only specifying queries for selecting tag-nodes, the "name" part within that node as well as the "value" part. This is necessary if you cannot or do not want to specifying them explicitly (e.g. if there is a large number or the tag-names are not known in advance).

The example given in PR #10576
```json
{
  "measurement": "measurementName",
  "timestamp": "2022-02-02T00:00:00Z",
  "tags": {
    "tag1": "tagValue1",
    "tag2": "tagValue2"
  },
  "fields": {
    "field1": 1.0,
    "field2": 2.0
  }
}
```
can then be parsed via
```toml
[[inputs.file]]
  files = ["example.json"]
  data_format = "xpath_json"

  [[inputs.file.xpath]]
    metric_name = "/measurement"
    timestamp = "/timestamp"
    timestamp_format = "2006-01-02T15:04:05Z"
    field_selection = "fields/child::*"
    tag_selection = "tags/child::*"
```

### CODE REVIEW

**Comment by wz2b** [comment]
If this gets merged, then it may eliminate my need for #10576.  The thing I've been looking for is something that can pull the tagset in dynamically; adding it to json_v2 is a little tricky because it allows everything to be recursively nested in the JSON then flattened out when it is transformed into a metric.  Maybe having a dynamic tag set should be considered a more customized or advanced solution and users wanting to do that should be directed to xpath_json instead.

Regardless, I'm expressing support for this PR, it would not only solve my problem but also provide a deserialization mechanism that can be used with https://github.com/influxdata/telegraf/tree/master/plugins/serializers/json which is really I am looking for.

File: config/config.go [hunk 1]
```diff
@@ -1609,8 +1609,12 @@ func (c *Config) getParserConfig(name string, tbl *ast.Table) (*parsers.Config,
 					c.getFieldStringMap(subtbl, "fields_int", &subcfg.FieldsInt)
 					c.getFieldString(subtbl, "field_selection", &subcfg.FieldSelection)
 					c.getFieldBool(subtbl, "field_name_expansion", &subcfg.FieldNameExpand)
-					c.getFieldString(subtbl, "field_name", &subcfg.FieldNameQuery)
-					c.getFieldString(subtbl, "field_value", &subcfg.FieldValueQuery)
+					c.getFieldString(subtbl, "field_name", &subcfg.TagNameQuery)
+					c.getFieldString(subtbl, "field_value", &subcfg.TagValueQuery)
```
Comment (powersj) [inline config/config.go:1613]: Are these two really supposed to be `tagnamequery` and `tagvaluequery` and not the corresponding `field` values?

Comment (srebhan) [reply config/config.go]: Very good catch @powersj! Fixed.

**Review by sspaink — APPROVED** [review]
Great work!

**Review by powersj — APPROVED** [review]

**Pull request merged**